crates/bevy_render/src/render_graph/nodes/render_resources_node.rs

```diff
@@ -808,3 +808,192 @@ where
+    // No need to do anything if no changes
+    if render_resources.is_changed() {
+        // Precalculate the aligned size of the whole render resources buffer
+        let aligned_size = render_resources
+            .iter()
+            .fold(0, |aligned_size, render_resource| {
+                aligned_size
+                    + render_resource_context
+                        .get_aligned_uniform_size(render_resource.buffer_byte_len().unwrap(), false)
+            });
+
+        // Get old buffer and possibly resize
+        let staging_buffer = if let Some(staging_buffer) = state.staging_buffer {
+            if render_resource_context
+                .get_buffer_info(staging_buffer)
+                .unwrap()
+                .size
+                != aligned_size
+            {
+                render_resource_context.remove_buffer(staging_buffer);
+                render_resource_context.create_buffer(BufferInfo {
+                    size: aligned_size,
+                    buffer_usage: BufferUsage::COPY_SRC | BufferUsage::MAP_WRITE,
+                    mapped_at_creation: true,
+                })
+            } else {
+                staging_buffer
+            }
+        } else {
+            // Or create a new one
+            render_resource_context.create_buffer(BufferInfo {
+                size: aligned_size,
+                buffer_usage: BufferUsage::COPY_SRC | BufferUsage::MAP_WRITE,
+                mapped_at_creation: true,
+            })
+        };
+
+        // Get old buffer and possibly resize
+        let target_buffer = if let Some(target_buffer) = state.target_buffer {
+            if render_resource_context
+                .get_buffer_info(target_buffer)
+                .unwrap()
+                .size
+                != aligned_size
+            {
+                render_resource_context.remove_buffer(target_buffer);
+                render_resource_context.create_buffer(BufferInfo {
+                    size: aligned_size,
+                    buffer_usage: BufferUsage::COPY_DST | BufferUsage::UNIFORM,
+                    mapped_at_creation: false,
+                })
+            } else {
+                target_buffer
+            }
+        } else {
+            // Or create a new one
+            render_resource_context.create_buffer(BufferInfo {
+                size: aligned_size,
+                buffer_usage: BufferUsage::COPY_DST | BufferUsage::UNIFORM,
+                mapped_at_creation: false,
+            })
+        };
+
+        // Write the resources into the staging buffer
+        let mut offset = 0;
+        for (index, render_resource) in render_resources.iter().enumerate() {
+            let render_resource_name = render_resources.get_render_resource_name(index).unwrap();
+            // dbg!(&render_resource_name);
+
+            let size = render_resource.buffer_byte_len().unwrap();
+            let aligned_size = render_resource_context.get_aligned_uniform_size(size, false);
```

Comment:
Instead of constantly casting `offset` and `aligned_sized` to a `u64`, just cast `aligned_sized` to a `u64` here. 

```rust
let aligned_size = render_resource_context.get_aligned_uniform_size(size, false) as u64;
```

Reply:
Win some, lose some, but I think it's still net positive.